sdk/keyvault/keyvault-keys/review/keyvault-keys.review.api.md

```diff
@@ -0,0 +1,295 @@
+    wrapKey(algorithm: KeyWrapAlgorithm, key: Uint8Array, options?: RequestOptions): Promise<WrapResult>;
+}
+
+// @public
+export interface DecryptOptions extends RequestOptions {
+    authenticationData?: Uint8Array;
+    authenticationTag?: Uint8Array;
+    iv?: Uint8Array;
+}
+
+// @public
+export interface DecryptResult {
+    algorithm: JsonWebKeyEncryptionAlgorithm;
+    keyID?: string;
+    result: Uint8Array;
+}
+
+// @public
+export interface DeletedKey extends Key {
+    readonly deletedDate?: Date;
+    readonly recoveryId?: string;
+    readonly scheduledPurgeDate?: Date;
+}
+
+// @public
+export type DeletionRecoveryLevel = "Purgeable" | "Recoverable+Purgeable" | "Recoverable" | "Recoverable+ProtectedSubscription";
+
+// @public
+export interface EncryptOptions extends RequestOptions {
+    authenticationData?: Uint8Array;
+    iv?: Uint8Array;
+}
+
+// @public
+export interface EncryptResult {
+    algorithm: JsonWebKeyEncryptionAlgorithm;
+    authenticationData?: Uint8Array;
+    authenticationTag?: string;
+    iv?: Uint8Array;
+    keyID?: string;
+    result: Uint8Array;
+}
+
+// @public
+export interface GetKeyOptions {
+    requestOptions?: msRest.RequestOptionsBase;
+    version?: string;
+}
+
+// @public
+export interface GetKeysOptions {
+    requestOptions?: msRest.RequestOptionsBase;
+}
+
+// @public
+export interface ImportKeyOptions {
+    enabled?: boolean;
+    expires?: Date;
+    hsm?: boolean;
+    notBefore?: Date;
+    requestOptions?: msRest.RequestOptionsBase;
+    tags?: {
+        [propertyName: string]: string;
+    };
+}
+
+// @public
+export interface JsonWebKey {
+    crv?: JsonWebKeyCurveName;
+    d?: Uint8Array;
+    dp?: Uint8Array;
+    dq?: Uint8Array;
+    e?: Uint8Array;
+    k?: Uint8Array;
+    // (undocumented)
+    keyOps?: string[];
+    kid?: string;
+    kty?: JsonWebKeyType;
+    n?: Uint8Array;
+    p?: Uint8Array;
+    q?: Uint8Array;
+    qi?: Uint8Array;
+    t?: Uint8Array;
+    x?: Uint8Array;
+    y?: Uint8Array;
+}
+
+// @public
+export type JsonWebKeyCurveName = "P-256" | "P-384" | "P-521" | "P-256K";
+
+// @public
+export type JsonWebKeyEncryptionAlgorithm = "RSA-OAEP" | "RSA-OAEP-256" | "RSA1_5";
+
+// @public
+export type JsonWebKeyOperation = "encrypt" | "decrypt" | "sign" | "verify" | "wrapKey" | "unwrapKey";
+
+// @public
+export type JsonWebKeyType = "EC" | "EC-HSM" | "RSA" | "RSA-HSM" | "oct";
+
+// @public
+export interface Key {
+    keyMaterial?: JsonWebKey;
+    properties: KeyProperties;
+}
+
+// @public
+export interface KeyProperties extends ParsedKeyVaultEntityIdentifier {
+    readonly created?: Date;
+    enabled?: boolean;
+    expires?: Date;
+    id?: string;
+    notBefore?: Date;
+    readonly recoveryLevel?: DeletionRecoveryLevel;
+    tags?: {
+        [propertyName: string]: string;
+    };
+    readonly updated?: Date;
+}
+
+// @public
+export class KeysClient {
```

Comment:
What do we think of `KeyClient` vs `KeysClient`? It seems odd to be plural

Reply:
It seemed weird to me at first but now I think it makes sense: it's the "keys" service, not the "key" service.  Same for "secrets" and "certificates"

Reply:
A blob client is a client for blobs :: A key client is a client for keys.

Reply:
Agreed.